### DISCUSSION TITLE
Adapting project from 4.11.0 to 5.3.1

### DISCUSSION THREAD

**Discussion by Lbones-ai** [HaxeFlixel/flixel#2859] [Q&A] [answered]
Hey guys, I recently update flixel from 4.11.0 to 5.3.1 and tried to run an old project, as expected I got some errors and warnings, but I don't get some of them:

- "Static access to instance field add is not allowed" for code:
  `FlxMouseEventManager.add(tempS, onMouseDownC, onMouseUpC, onMouseOverC, onMouseOutC, true);`
I checked the API but they didn't mention this, they explicitly encourage this approach: https://api.haxeflixel.com/flixel/input/mouse/FlxMouseEventManager.html

- "Warning : Invalid name: 1_resource__png for file: assets/images/1_resource.png"

Can someone help me?
thanks!

**Comment by Geokureli** [comment, marked as answer]
I recommend upgrading to flixel 5.0.0 first, using [this 5.0.0 migration guide](https://github.com/HaxeFlixel/flixel/wiki/Flixel-5.0.0-Migration-guide). Once you've done that, then you should be able to jump to 5.3.1.

as for you're specific error, it's outlined [here](https://github.com/HaxeFlixel/flixel/wiki/Flixel-5.0.0-Migration-guide#flxmouseeventmanager-is-now-an-instance-not-a-static-manager-2540) in the migration guide, and the fix is simple: Replace static calls like `FlxMouseEventManager.add(...)` with calls to `FlxMouseEvent.add(...)`. We added a default mouse event manager, similarly to how FlxTween and FlxTimer work.

Once you're on 5.0.0, you should only have deprecation warnings, and you can either ask about them here or check the [changelog](https://github.com/HaxeFlixel/flixel/blob/dev/CHANGELOG.md) for details

**Reply by Lbones-ai** [reply]
Thank you, I guess the API it's out of date then

**Reply by Geokureli** [reply]
https://api.haxeflixel.com/flixel/input/mouse/FlxMouseEventManager.html
Yeah, looks like it :(
I'll fix that
